tests/ceph_ansible/test_ansible_upgrade.py

```diff
@@ -213,7 +218,38 @@ def run(ceph_cluster, **kw):
             LOG.error("Failed during cephadm adopt (rc = {})".format(rc))
             return rc
 
+        LOG.info("The value for parameter build is {}".format(build))
+        if build.startswith("5.1"):
+            installer = ceph_cluster.get_nodes(role="installer")[0]
+            base_cmd = "sudo cephadm shell -- ceph"
+            config_cmd = f"{base_cmd} config set mgr mgr/cephadm/yes_i_know true"
+            installer.exec_command(cmd=config_cmd)
+            mgr_cmd = f"{base_cmd} mgr fail"
+            installer.exec_command(cmd=mgr_cmd)
+
         client = ceph_cluster.get_nodes("mon")[0]
+        if config.get("verify_cephadm_containers"):
+            for node in ceph_cluster.get_nodes(ignore="client"):
+                out, err = node.exec_command(cmd="cephadm ls", sudo=True)
+                legacy_services = [x for x in loads(out) if x["style"] == "legacy"]
+                if legacy_services:
+                    LOG.info(
+                        f"Legacy services present in cephadm node {node.hostname} after adopting the cluster"
+                    )
+                    LOG.info(
+                        "Checking cluster status to ensure that the legacy services are not being inferred"
+                    )
+                    rc = ceph_cluster.check_health(
+                        build,
+                        cluster_name=cluster_name,
+                        client=client,
+                        timeout=config.get("timeout", 300),
+                    )
+                    if rc != 0:
+                        LOG.error(
+                            "Ceph health not OK after adopting cluster to use cephadm"
+                        )
+                        return rc
```

Comment:
Can this be a separate check?

Reply:
Separate check outside test_ansible_upgrade.py module?

Reply:
Done.